test/solvers.jl

```diff
@@ -102,3 +102,8 @@ minlp_solvers = Any[]
 kni && push!(minlp_solvers, KNITRO.KnitroSolver(outlev=0))
 osl && push!(minlp_solvers, CoinOptServices.OsilBonminSolver())
 osl && push!(minlp_solvers, CoinOptServices.OsilCouenneSolver())
+
+using Compat
+const error_map = @compat Dict()
+grb && (error_map[Gurobi.GurobiSolver] = Gurobi.GurobiError)
+cpx && (error_map[CPLEX.CplexSolver] = CPLEX.CplexError)
```

Comment:
I haven't touched that code yet, I wasn't sure if the error codes would be passed through properly.